core/corehttp/gateway_handler.go

```diff
@@ -159,7 +159,12 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 
 	nd, err := core.Resolve(ctx, i.node, path.Path(urlPath))
-	if err != nil {
+	// If node is in offline mode the error code and message should be different
+	if err == core.ErrNoNamesys && !i.node.OnlineMode() {
```

Comment:
i think there's other errors here to check alongside `!i.node.OnlineMode()`. for example, if a `/ipfs/...` path fails to resolve, it's likely because of no network.